### DISCUSSION TITLE
Should Beaker parse YAML front matter?

### DISCUSSION THREAD

**Discussion by johanbove** [beakerbrowser/beaker#1649] [General]
In the last resolved Markdown related issue (#1610 ) Paul brought parsing of YAML Front Matter (YFM) up.

I personally use YFM in a lot of my markdown files to add meta information to the files. Some of these files were copied out of my old Jekyll blog.

If Beaker could handle this "front matter" and not render it in the final HTML, but parse it the way Jekyll, Hugo and other CMSes do, this might help other website owners migrate over to Hyperdrive more easily.

Of course it is up for discussion as to just how much parsing Beaker should be doing here?

I would suggest support for these tags: title, status, description, author, tag, category

We could also add control over how the file is parsed to HTML by adding the keys: layout, published, etc

## Resources
- <https://jekyllrb.com/docs/front-matter/>
- <https://assemble.io/docs/YAML-front-matter.html>
- <https://gohugo.io/content-management/front-matter/>

**Comment by pfrazee** [comment]
I saw Matthew Auld send this via email and it's not clear to me whether they deleted the comment or GH discussions bugged out (which it has done in the past) so I'm reproing here:

> On suggestion might be to move the YFM to [file metadata](https://docs.beakerbrowser.com/advanced/editing-file-metadata) - each file has key: value metadata that can be totally user defined. It wouldn't be too much trouble to write a webterm command that could batch-parse files and copy it's YMF to the metadata and remove it from file content.

> You could also take a look at the frontend feature, where you entirely customize how a url on a given drive it displayed.

**Comment by zootella** [comment]
@johanbove @pfrazee 

Jekyll/Hugo/Hexo and others do follow the pattern of:
* 1 post per file;
* YAML at the top between `---` lines;
* Markdown after that.

They then use a huge CLI preprocessor to convert all that into your static site.

I have had success parsing YAML and Markdown right in the browser, though, using (possibly) the same modules they are:

http://nodeca.github.io/js-yaml/
https://www.npmjs.com/package/markdown-it
https://www.npmjs.com/package/highlight.js

Ironically, the YAML module was only made to work in the browser to power the demonstration on its webpage. But the parsing is just as good.

**Reply by zootella** [reply]
https://github.com/zootella/modules

^more links to stuff I've found that works well on pages delivered with `http://`, `file://`, and/or `hyper://`. With [Babel Standalone](https://babeljs.io/docs/en/next/babel-standalone.html), for instance, you can get React in Beaker.
